src/test/java/codeu/controller/RegisterServletTest.java

```diff
@@ -4,31 +4,92 @@
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
+import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
 
-public class RegisterServletTest
+import codeu.model.data.User;
+import codeu.model.store.basic.UserStore;
+
+public class RegisterServletTest 
 {
-  private RegisterServlet registerServlet;
-  private HttpServletRequest mockRequest;
-  private PrintWriter mockPrintWriter;
-  private HttpServletResponse mockResponse;
+
+	private RegisterServlet registerServlet;
+	private HttpServletRequest mockRequest;
+	private PrintWriter mockPrintWriter;
+	private HttpServletResponse mockResponse;
+
+	@Before
+	public void setup() throws IOException 
+	{
+		registerServlet = new RegisterServlet();
+		mockRequest = Mockito.mock(HttpServletRequest.class);
+		mockPrintWriter = Mockito.mock(PrintWriter.class);
+		mockResponse = Mockito.mock(HttpServletResponse.class);
+		Mockito.when(mockResponse.getWriter()).thenReturn(mockPrintWriter);
+	}//setup
+
+	@Test
+	public void testDoGet() throws IOException, ServletException
+	{
+		registerServlet.doGet(mockRequest, mockResponse);
+		Mockito.verify(mockPrintWriter).println("<h1>RegisterServlet GET request.</h1>");
+	}//testDoGet
  
-  @Before
-  public void setup() throws IOException 
-  {
-    registerServlet = new RegisterServlet();
-    mockRequest = Mockito.mock(HttpServletRequest.class);
-    mockPrintWriter = Mockito.mock(PrintWriter.class);
-    mockResponse = Mockito.mock(HttpServletResponse.class);
-    Mockito.when(mockResponse.getWriter()).thenReturn(mockPrintWriter);
-  }//setup
-
-  @Test
-  public void testDoGet() throws IOException, ServletException
-  {
-    registerServlet.doGet(mockRequest, mockResponse);
-    Mockito.verify(mockPrintWriter).println("<h1>RegisterServlet GET request.</h1>");
-  }//testDoGet
+	@Test
+	public void testDoPost_BadUsername() throws IOException, ServletException {
+		Mockito.when(mockRequest.getParameter("username")).thenReturn("bad !@#$% username");
+
+		loginServlet.doPost(mockRequest, mockResponse);
```

Comment:
I don't think this is quite right. 'loginServlet' should instead be 'registerServlet'. The reason this test appeared to pass before, was because you have the old code for LoginServlet. LoginServlet has now been updated to not have a check for a bad username since it is handled here. Execute the command 'git pull origin master' to get the latest changes. And quick tip for the future: RegisterServletTest is **only** supposed to test stuff from RegisterServlet. Leave the testing of methods in LoginServlet to the class LoginServletTest.